torch/utils/data/dataloader.py

```diff
@@ -195,8 +202,9 @@ def __init__(self, loader):
             self.workers = [
                 multiprocessing.Process(
                     target=_worker_loop,
-                    args=(self.dataset, self.index_queue, self.worker_result_queue, self.collate_fn))
-                for _ in range(self.num_workers)]
+                    args=(self.dataset, self.index_queue, self.worker_result_queue, self.collate_fn,
+                          self.worker_init_fn, i))
```

Comment:
Can you check if this works with `spawn` start method? I'm not sure if you can serialize arbitrary lambdas like this.